src/components/Modals/Tracks/CreateTrack.jsx

```diff
@@ -1,136 +1,70 @@
 import PropTypes from 'prop-types';
-import React from 'react';
 import classes from './CreateTrack.module.css';
 import Button from '../../Button/Button';
 import CraftInput from '../CraftInput';
 
 import { TRACKS, createElemRefOnDB } from '../../../utilities/fb-helpers';
-import {
-  CREATE_TRACK_ONCHANGE,
-  CREATE_TRACK_VALIDATE_FIELDS,
-  CREATE_TRACK_VALIDATE_FORM,
-  reducerFunc,
-} from './track-helpers';
-import debounce from '../../../utilities/debounce';
+import { useInput, useValidator } from '../modals-helpers';
 
-export default class CreateTrack extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      title: '',
-      startDate: '',
-      data: {
-        id: createElemRefOnDB(TRACKS).id,
-        name: '',
-        date: '',
-        note: '',
-      },
-      validator: {
-        name: false,
-        date: false,
-      },
-      errors: {
-        nameError: '',
-        dateError: '',
-      },
-      isValid: false,
-    };
-    this.liftUpCreateTrack = this.liftUpCreateTrack.bind(this);
-    this.closeModal = this.closeModal.bind(this);
-    this.onChange = this.onChange.bind(this);
-    this.validateFields = this.validateFields.bind(this);
-    this.validateForm = this.validateForm.bind(this);
-  }
+export default function CreateTrack({ task: { title, startDate }, closeFunc, liftUpCreateTrack }) {
+  const { state, onChange } = useInput({
+    id: createElemRefOnDB(TRACKS).id,
```

Comment:
looks like your method can be called outside of the component and you will avoid extra work

Reply:
I created a ref: ` const newTrackRef = useRef(createElemRefOnDB(TRACKS));`
```suggestion
  const { state, onChange } = useInput(() => ({
    id: newTrackRef.current.id,
    name: '',
    date: '',
    note: '',
  }));
```